packages/frontend/src/components/Game/GameBoard/TileGrid/index.jsx

```diff
@@ -12,20 +12,18 @@ const TileGrid = ({ tiles }) => {
   };
 
   useEffect(() => {
-    if (state.flipped.length > 1) {
-      const timeout = setTimeout(() => {
-        if (state.flipped[0].photo.id === state.flipped[1].photo.id) {
-          dispatch(handleMatch(state.flipped));
+    const handleFlippedPair = (flipped) => {
```

Comment:
If we define handleFlippedPair in the actions file, it will not have access to the contexts dispatch function which needs to call handle match and reset tiles.

Reply:
`dispatch` could be passed as an argument to the `handleMatch` action, but I think it's okay to leave it here.

This could also be a good use case for currying, I believe.

```javascript
const handleMatch = () => {...}
const resetTiles = () => {...}

export const handleMatchDispatcher = (dispatch) => (tile) => {
  setTimeout(() => {
    if(flipped[0].photo.id === flipped[1].photo.id){
      dispatch(handleMatch(tile))
    } else {
      dispatch(resetTiles())
    }
  }, 2000)
}
```

Then when it was used in the component:
```javascript
import { useGameContext } from 'gameContext';
import { handleMatchDispatcher } from 'actions';

const MyComponent = () => {
  const { dispatch } = useGameContext();
  const handleMatch = handleMatchDispatcher(dispatch);

  const onClick = (tile) => {
    handleMatch(tile);
  }
}
```